library/src/scala/util/Properties.scala

```diff
@@ -54,11 +54,11 @@ private[scala] trait PropertiesTrait {
 
   def propIsSet(name: String)                   = System.getProperty(name) != null
   def propIsSetTo(name: String, value: String)  = propOrNull(name) == value
-  def propOrElse(name: String, alt: => String)  = Option(System.getProperty(name)).getOrElse(alt)
+  def propOrElse(name: String, alt: => String | Null)  = Option(System.getProperty(name)).getOrElse(alt)
```

Comment:
There are similar functions within the compiler that were ported to explicit nulls in #23368 . Consider making those changes consistent, either by changing this to be the same as in the compiler, or changing the compiler to be the same as this.